File: translateRotateScaleAlphaSetAnimation/app/src/main/java/com/example/translaterotatescalealphasetanimation/MainActivity.java
package com.example.translaterotatescalealphasetanimation;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    Button buttonTranslate,buttonRotate,buttonScale,buttonAlpha,buttonSet;
    ImageView imageAnimation;
    Animation animTranslate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initialize();
    }


    private void initialize(){

        buttonTranslate = findViewById(R.id.button_translate);
        buttonRotate = findViewById(R.id.button_rotate);
        buttonScale = findViewById(R.id.button_scale);
        buttonAlpha = findViewById(R.id.button_alpha);
        buttonSet = findViewById(R.id.button_set);
        imageAnimation = findViewById(R.id.image_animation);

        buttonTranslate.setOnClickListener(this);
        buttonRotate.setOnClickListener(this);
        buttonScale.setOnClickListener(this);
        buttonAlpha.setOnClickListener(this);
        buttonSet.setOnClickListener(this);

        animTranslate = AnimationUtils.loadAnimation(this,R.anim.translate);

    }

    @Override
    public void onClick(View v) {

        switch (v.getId()){
            case R.id.button_translate:
                imageAnimation.startAnimation(animTranslate);
                break;

            case R.id.button_rotate:

                break;

            case R.id.button_scale:

                break;

            case R.id.button_alpha:

                break;

            case R.id.button_set:

                break;

        }

    }
}

File: volley/volley2/app/src/main/java/com/example/volley/MainActivity.kt
package com.example.volley

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import org.json.JSONException

class MainActivity : AppCompatActivity() {
    private var mTextViewResult: TextView? = null
    private var mQueue: RequestQueue? = null
    private var buttonParse:Button?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mTextViewResult = findViewById(R.id.text_view_result)
        buttonParse = findViewById(R.id.button_parse)
        mQueue = Volley.newRequestQueue(this)
        buttonParse?.setOnClickListener { jsonParse() }
    }

    private fun jsonParse() {
        val url = "https://pixabay.com/api/?key=5303976-fd6581ad4ac165d1b75cc15b3&q=kitten&image_type=photo&pretty=true"
        val request = JsonObjectRequest(Request.Method.GET, url, null,
                Response.Listener { response ->
                    try {
                        val jsonArray = response.getJSONArray("hits")
                        for (i in 0..jsonArray.length()) {
                            val employee = jsonArray.getJSONObject(i)
                            val firstname = employee.getString("tags")
                            val age = employee.getInt("likes")
                            val mail = employee.getString("type")
                            mTextViewResult!!.append("$firstname, $age, $mail\n\n")
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                }, Response.ErrorListener { error -> error.printStackTrace() })
        mQueue!!.add(request)
    }
}